src/main/java/me/zeroeightsix/kami/command/commands/ConfigCommand.kt

```diff
@@ -1,77 +1,160 @@
+                    execute("Copy current preset to specific preset") {
+                        val name = nameArg.value
+                        if (!confirm(name)) return@execute
+
+                        configTypeArg.value.copyPreset(name)
+                    }
+                }
+            }
+
+            literal("delete", "del", "remove") {
+                string("name") { nameArg ->
+                    execute("Delete specific preset") {
+                        val name = nameArg.value
+                        if (!confirm(name)) return@execute
+
+                        configTypeArg.value.deletePreset(name)
                     }
                 }
             }
 
-            execute("Print current config files") {
-                defaultScope.launch(Dispatchers.IO) {
-                    val path = Paths.get(ModuleConfig.currentPath).toAbsolutePath()
-                    MessageSendHelper.sendChatMessage("Path to configuration: ${formatValue(path)}")
+            literal("list") {
+                execute("List all available presets") {
+                    configTypeArg.value.printAllPresets()
+                }
+            }
+
+            literal("server") {
+                literal("create", "new", "add") {
+                    executeSafe("Create a new server preset") {
+                        val ip = getIpOrNull() ?: return@executeSafe
+
+                        configTypeArg.value.newServerPreset(ip)
+                    }
+                }
+
+                literal("delete", "del", "remove") {
+                    executeSafe("Delete the current server preset") {
+                        val ip = getIpOrNull() ?: return@executeSafe
+                        val configType = configTypeArg.value
+
+                        if (!configType.serverPresets.contains(ip)) {
+                            MessageSendHelper.sendChatMessage("This server doesn't have a preset in config ${configType.displayName}")
+                            return@executeSafe
+                        }
+
+                        if (!confirm(ip)) return@executeSafe
+
+                        configType.deleteServerPreset(ip)
+                    }
+                }
+
+                literal("list") {
+                    execute("List all available server presets") {
+                        configTypeArg.value.printAllServerPreset()
+                    }
                 }
             }
+
+            execute("Print current preset name") {
+                configTypeArg.value.printCurrentPreset()
+            }
+        }
+    }
+
+    private fun SafeExecuteEvent.getIpOrNull(): String? {
+        val ip = mc.currentServerData?.serverIP
+
+        return if (ip == null || mc.isIntegratedServerRunning) {
+            MessageSendHelper.sendWarningMessage("You are not in a server!")
+            null
+        } else {
+            ip
+        }
+    }
+
+    private fun confirm(name: String): Boolean {
```

Comment:
You're only using the arg name without the literal itself, meaning if someone does
```
;config copy config1 // the user will be asked to confirm deletion
;config delete config1 // the user will not be asked to confirm deletion
```
the confirm name should include the literal arg name